api/pkg/transformer/server/grpc/server.go

```diff
@@ -0,0 +1,249 @@
+package grpc
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+	"time"
+
+	hystrixpkg "github.com/gojek/merlin/pkg/hystrix"
+	"github.com/gojek/merlin/pkg/transformer/pipeline"
+	"github.com/gojek/merlin/pkg/transformer/server/config"
+	"github.com/gojek/merlin/pkg/transformer/server/grpc/interceptors"
+	"github.com/gojek/merlin/pkg/transformer/server/instrumentation"
+	"github.com/gojek/merlin/pkg/transformer/types"
+	"github.com/opentracing/opentracing-go"
+	"github.com/pkg/errors"
+
+	"github.com/afex/hystrix-go/hystrix"
+	upiv1 "github.com/caraml-dev/universal-prediction-interface/gen/go/grpc/caraml/upi/v1"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/reflection"
+	"google.golang.org/grpc/status"
+)
+
+var (
+	shutdownSignals = []os.Signal{os.Interrupt, syscall.SIGTERM}
+)
+
+// UPIServer serves GRPC request that implement UPI Service
+type UPIServer struct {
+	upiv1.UnimplementedUniversalPredictionServiceServer
+
+	opts        *config.Options
+	modelClient upiv1.UniversalPredictionServiceClient
+	conn        *grpc.ClientConn
+	logger      *zap.Logger
+
+	// ContextModifier function to modify or store value in a context
+	ContextModifier func(ctx context.Context) context.Context
+	// PreprocessHandler function to run all preprocess operation
+	PreprocessHandler func(ctx context.Context, request types.Payload, requestHeaders map[string]string) (types.Payload, error)
+	// PostprocessHandler function to run all preprocess operation
```

Comment:
Typo, should be `postprocess` . Can you also add comment that explains input and output of both `PreprocessHandler` and `PostprocessHandler` ?